kmk/modules/split.py

```diff
@@ -84,22 +85,25 @@ def during_bootup(self, keyboard):
             if not self.data_pin:
                 self.data_pin = keyboard.data_pin
 
-        # Detect split side from name
-        if self.split_side is None:
-            if name.endswith('L'):
-                # If name ends in 'L' assume left and strip from name
-                self._is_target = bool(self.split_target_left)
-                self.split_side = SplitSide.LEFT
-            elif name.endswith('R'):
-                # If name ends in 'R' assume right and strip from name
-                self._is_target = not bool(self.split_target_left)
-                self.split_side = SplitSide.RIGHT
-
         # if split side was given, find master from split_side.
-        elif self.split_side == SplitSide.LEFT:
+        if self.split_side == SplitSide.LEFT:
             self._is_target = bool(self.split_target_left)
         elif self.split_side == SplitSide.RIGHT:
             self._is_target = not bool(self.split_target_left)
+        else:
+            # Detect split side from name
+            if (
+                self.split_type == SplitType.UART
+                or self.split_type == SplitType.ONEWIRE
+            ):
+                self._is_target = runtime.usb_connected
+            elif self.split_type == SplitType.BLE:
+                self._is_target = bool(self.split_target_left)
```

Comment:
I don't have any BLE controllers to test if it will work, but I see here a change which potentially could break BLE. If split_side is not declared, but recognized from drive name, both BLE sides will get the same _is_target value instead of pair of opposite values. Am I missing something?
```suggestion
                self._is_target = name.endswith('L') == self.split_target_left
```

Reply:
My code replicates the existing functionality. As far as I can see you arent missing anything. Im not not sure what to do in this case. Your code assumes that the ble user will want ee hands, but what happens if they dont have the drives labelled?

Reply:
My code replicates the existing functionality. As far as I can see you arent missing anything. Im not not sure what to do in this case. Your code assumes that the ble user will want ee hands, but what happens if they dont have the drives labelled?

Reply:
Honestly I think the better thing to do is test for these things when initializing and bail out with an error if nothing is set.

Reply:
now that I can look at this on a bigger screen, Your suggestion makes sense. I just couldnt figure out what was going on there.